src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -175,6 +182,47 @@ void RaftReplService::stop() {
     hs()->logstore_service().stop();
 }
 
+void RaftReplService::monitor_cert_changes() {
+    auto fw = ioenvironment.get_file_watcher();
+    auto cert_change_cb = [this](const std::string filepath, const bool deleted) {
+        LOGINFO("file change event for {}, deleted? {}", filepath, deleted)
+        // do not block file_watcher thread
+        std::thread restart_svc(&RaftReplService::restart_raft_svc, this, filepath, deleted);
+        restart_svc.detach();
+    };
+
+    //monitor ssl cert file
+    if (!fw->register_listener(ioenvironment.get_ssl_cert(), "hs_ssl_cert_watcher", cert_change_cb)) {
+        LOGERROR("Failed to register listner, {} to watch file {}, Not monitoring cert files",
+                   "hs_ssl_cert_watcher", ioenvironment.get_ssl_cert());
+    }
+    //monitor ssl key file
+    if (!fw->register_listener(ioenvironment.get_ssl_key(), "hs_ssl_key_watcher", cert_change_cb)) {
+        LOGERROR("Failed to register listner, {} to watch file {}, Not monitoring cert files",
+                   "hs_ssl_key_watcher", ioenvironment.get_ssl_key());
+    }
+}
+
+void RaftReplService::restart_raft_svc(const std::string filepath, const bool deleted){
```

Comment:
this method will be called when cert or key is deleted. that means when key is deleted we restart once , when cert is deleted we restart once.

will there be a moment that a new key exists with a old cert?  will this bring any issue if we restart with a new key and old cert? I mean can we restart raft svc only when both key and cert is updated?

pls double check. if there is no harm to restart with a new key and old cert , then I thinks it will be good  since raft svc will be restarted soon when the new cert is updated.

also, in monitor_cert_changes, we will register listener for both key and cert. if we restart with the update of key, we will register listener for cert again. pls also double check if this is good